NotificationService Отправка Email сообщений BJS2 41877 redis email

SerUKalinin opened the pull request.

SerUKalinin committed:
EmailService

SerUKalinin committed:
EmailServiceTest

SerUKalinin committed:
FollowerEvent

SerUKalinin committed:
FollowerEventListener

SerUKalinin committed:
FollowerMessageBuilder

SerUKalinin committed:
messages.properties

SerUKalinin committed:
RedisConfig

abasilashvili commented on src/main/java/faang/school/notificationservice/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,58 @@
+        RedisStandaloneConfiguration redisStandaloneConfiguration = new RedisStandaloneConfiguration(redisHost, redisPort);
+        return new LettuceConnectionFactory(redisStandaloneConfiguration);
+    }
+
```
Поменял библиотеку на Lettuce, в названии метода оставил jedis, в итоге Lettuce, jedis, Redis в одном методе выглядит некрасиво. jedisConnectionFactory поменял бы на lettuceConnectionFactory

abasilashvili commented on src/main/java/faang/school/notificationservice/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,58 @@
+        redisTemplate.setKeySerializer(new StringRedisSerializer());
+        return redisTemplate;
+    }
+
```
тут тоже

abasilashvili commented on src/main/java/faang/school/notificationservice/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,58 @@
+    ChannelTopic topic() {
+        return new ChannelTopic("follower_channel");
+    }
+
```
Можешь follower_channel вынести в yaml config, как порт, и хост. Если придется менять название изменишь только в конфиге. Название topic тоже поменял бы, потому что к Redis Config могут добавлятся другие топики. Например followerChannelTopic думаю норм

abasilashvili commented on src/main/java/faang/school/notificationservice/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,58 @@
+    @Bean
+    RedisMessageListenerContainer redisContainer(MessageListenerAdapter followerMessageListenerAdapter) {
+        RedisMessageListenerContainer container = new RedisMessageListenerContainer();
+        container.setConnectionFactory(jedisConnectionFactory());
```
jedis.

abasilashvili commented on src/main/java/faang/school/notificationservice/config/context/RedisConfig.java:
```diff
@@ -0,0 +1,58 @@
+    RedisMessageListenerContainer redisContainer(MessageListenerAdapter followerMessageListenerAdapter) {
+        RedisMessageListenerContainer container = new RedisMessageListenerContainer();
+        container.setConnectionFactory(jedisConnectionFactory());
+        container.addMessageListener(followerMessageListenerAdapter, topic());
```
topic

SerUKalinin committed:
RedisConfig кросс ревью v1.1

SerUKalinin committed:
RedisConfig кросс ревью v1.1

SerUKalinin committed:
Доработка сервиса FollowerEvent

SerUKalinin committed:
Доработка сервиса FollowerEventListener

SerUKalinin committed:
Доработка сервиса FollowerMessageBuilder

SerUKalinin committed:
Доработка сервиса LocalDateTimeArrayDeserializer

SerUKalinin committed:
Доработка сервиса RedisConfig

SerUKalinin committed:
Доработка сервиса UserDto

SerUKalinin committed:
Доработка сервиса RedisConfig

SerUKalinin committed:
Доработка сервиса JacksonConfig

SerUKalinin committed:
Доработка сервиса LocalDateTimeArrayDeserializer

SerUKalinin committed:
Доработка сервиса RedisConfig

SerUKalinin committed:
Доработка сервиса перемещение LocalDateTimeArrayDeserializer в правильный пакет

SerUKalinin committed:
Доработка сервиса отправка на почту (готово)

SerUKalinin committed:
Доработка сервиса отправка на почту
Добавление выбора:
Выберите способ отправки уведомления:
1. EMAIL
2. SMS
3. TELEGRAM

SerUKalinin committed:
перенос топика в конфигурацию

SerUKalinin committed:
коммит

SerUKalinin commented:
![photo_2024-12-04_14-04-39](https://github.com/user-attachments/assets/8d7ea2ce-4684-4259-bad4-8cf69c3c8124)



каеф, пришло письмецо))

SerUKalinin committed:
Добавил MailController

SerUKalinin committed:
Изменил в UserDTO name на username

SerUKalinin commented:
![photo_2024-12-04_17-37-18](https://github.com/user-attachments/assets/593e1e73-20af-4ff0-b96d-0d8719a4e523)


каеф, работает))

SerUKalinin committed:
Тест

SerUKalinin committed:
@Slf4j

SerUKalinin committed:
доработка EmailService

SerUKalinin committed:
тест

zaqqaz226 commented:
ВСе ок, но не закатываю, так как жду ответа от команды по настройкам редиса
